internal/dataplane/parser/parser.go

```diff
@@ -33,35 +35,92 @@ import (
 
 const (
 	KindGateway = gatewayv1beta1.Kind("Gateway")
+
+	// kongRouterFlavorExpressions is the value used in router_flavor of kong configuration
+	// to enable expression based router of kong.
+	kongRouterFlavorExpressions = "expressions"
 )
 
 // -----------------------------------------------------------------------------
 // Parser - Public Types
 // -----------------------------------------------------------------------------
 
+// FeatureFlags are used to control the behavior of the parser.
+type FeatureFlags struct {
+	// ReportConfiguredKubernetesObjects turns on object reporting for this parser:
+	// each subsequent call to BuildKongConfig() will track the Kubernetes objects which
+	// were successfully parsed.
+	ReportConfiguredKubernetesObjects bool
+
+	// CombinedServiceRoutes changes the translation logic from the legacy
+	// mode which would create a kong.Route object per each individual path on
+	// an Ingress object to a mode that can combine routes for paths where the
+	// service name, host and port match for those paths.
+	CombinedServiceRoutes bool
+
+	// RegexPathPrefix enables adding the Kong 3.x+ regex path prefix on regex paths generated by the controller
+	// (to satisfy the Ingress Prefix and Exact path types) or indicated by a resource (e.g. when an HTTPRoute uses a
+	// RegularExpression Match). It does _not_ enable heuristic regex path detection for Ingress ImplementationSpecific
+	// paths, which require an IngressClass setting.
+	RegexPathPrefix bool
+
+	// ExpressionRoutes indicates whether to translate Kubernetes objects to expression based Kong Routes.
+	ExpressionRoutes bool
+}
+
+func NewFeatureFlags(
+	logger logrus.FieldLogger,
```

Comment:
could we consider replacing `logrus.FieldLogger` to `logr.Logger` in some future refactoring PRs? (not necessarily in this PR)